backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/optimizations/DFGBuilder.kt

```diff
@@ -230,7 +230,8 @@ internal class ModuleDFGBuilder(val context: Context, val irModule: IrModuleFrag
                         println("Analysing global field ${declaration.descriptor}")
                         println("IR: ${ir2stringWhole(declaration)}")
                     }
-                    analyze(declaration, IrSetFieldImpl(it.startOffset, it.endOffset, declaration.symbol, null,
+                    if (it.expression !is IrConst<*>)
```

Comment:
The entire `visitField` looks intended for global fields. Maybe adjust the entry condition instead then?
Otherwise debug output will be incorrect.

Reply:
Yeah, here implicitly being used the fact of erasing of all initializers for non-global fields. Probably better to check that explicitly by reading the field's owner and skipping if it's a class.